cmd/crank/describe.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/exp/slices"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/crossplane/crossplane/cmd/crank/printer"
+	"github.com/crossplane/crossplane/internal/k8s"
+)
+
+const (
+	errGetResource = "Couldn't get requested resource."
+	errCliOutput   = "Error printing CLI table."
+	errGraphOutput = "Error generating graph PNG."
+)
+
+// describeCmd describes a Kubernetes Crossplane resource.
+type describeCmd struct {
+	Kind       string   `arg:"" required:"" help:"Kind of resource to describe."`
+	Name       string   `arg:"" required:"" help:"Name of specified resource to describe."`
+	Namespace  string   `short:"n" name:"namespace" help:"Namespace of resource to describe." default:"default"`
+	Output     string   `short:"o" name:"output" help:"Output type of graph. Possible output types: tree, table, graph." enum:"tree,table,graph" default:"tree"`
+	Fields     []string `short:"f" name:"fields" help:"Fields that are printed out in the header." default:"kind,name"`
+	OutputPath string   `short:"p" name:"path" help:"Output path for graph PNG. Only valid when set with output=graph." default:"graph.png"`
+}
+
+func (c *describeCmd) Run(logger logging.Logger) error {
+	logger = logger.WithValues("Kind", c.Kind, "Name", c.Name)
+
+	allowedFields := []string{"parent", "name", "kind", "namespace", "apiversion", "synced", "ready", "message", "event"}
```

Comment:
I would suggest moving this to be a global variable

Reply:
Done